# The *Dopamine Dump* Theory

<img src="img/nba-logo.jpg" align="right" width="200"/>
<img src="img/brain.jpg" align="right" width="200"/>


The *dopamine dump* theory is a theory about how an NBA team performs on the second leg of a back-to-back after having won a close game the night before. The naming comes from the idea that, if you win a close game, you have a temporary high from winning and the next game is kind of a let-down game (shout out to the **[Great Western Podcast](https://www.youtube.com/watch?v=BUPT5r5P7os&t=1943s)** for coining the term). Using my go-to source for historical odds ([yahoo API](https://sports.yahoo.com/nba/)), let's test our theory by running numbers on how such a team performs **against the spread** and **straight up** in the previous years.

In [2]:
%run nba

what is today's date? 
 2025-11-05


2025-11-04 TOR MIL Final
2025-11-04 CHI PHI Final
2025-11-04 NO CHA Final
2025-11-04 ATL ORL Final
2025-11-04 GS PHO Final
2025-11-04 LAC OKC Final
2025-11-05 IND BKN Pregame
2025-11-05 DET UTA Pregame
2025-11-05 CLE PHI Pregame
2025-11-05 NY MIN Pregame
2025-11-05 BOS WAS Pregame
2025-11-05 MEM HOU Pregame
2025-11-05 DAL NO Pregame
2025-11-05 DEN MIA Pregame
2025-11-05 SAC GS Pregame
2025-11-05 POR OKC Pregame
2025-11-05 LAL SA Pregame


In [3]:
%run code

Gathering odds data
Forward filling
..making columns and cleaning the data...


 now working on the totals prediction columns ...
Predictions:
Predictions home / road adjusted predictions:
Basic predictions:
                matchup  yahoo_total  yahoo_spread  predicted_total  \
date                                                                  
2025-11-05  IND vs. BKN       231.50         -6.50           233.50   
2025-11-05  DET vs. UTA       232.50         -9.50           229.50   
2025-11-05  CLE vs. PHI       236.50        -10.50           233.00   
2025-11-05  BOS vs. WAS       232.50        -12.50           228.00   
2025-11-05  NYK vs. MIN       227.50         -3.50           231.00   
2025-11-05  MEM vs. HOU       232.50          7.50           232.50   
2025-11-05  DAL vs. NOP       227.50         -8.50           226.00   
2025-11-05  DEN vs. MIA       241.50         -9.50           238.00   
2025-11-05  SAC vs. GSW       226.50         -2.50           233.50   
2025-11-05 

To define *winning a close game*, I choose a game that finishes with a team winning by $1$ to $5$ points.

In [14]:
A = df['pts_diff_prev'].isin([1,2,3,4,5])
B = df['rest_numeric'] == 1
C = ~(df['pts_diff_prev_opp'].isin([1,2,3,4,5]) & df['rest_numeric_opp']==1)

In [15]:
b2b_win_avg = df.loc[B]['win'].mean()
filter_win_avg = df.loc[A&B&C,'win'].mean()
print("In back-to-backs:",round(b2b_win_avg,3),"%")
print("After close win and b2b:",round(filter_win_avg,3),"%")

In back-to-backs: 0.439 %
After close win and b2b: 0.391 %


Let's now see how this performs against the spread when we *bet against* a team in this situation. 

In [16]:
df.loc[A&B&C][['hundred_spread_fade','hundred_ml_fade']].agg(['count','sum','mean'])

hundred_spread_fade  hundred_ml_fade
count               457.00           457.00
sum                3675.83          2105.93
mean                  8.04             4.61

Following this theory, your ROI against the spread is $8$%, and $4$% on the moneyline. Not bad for a sample size of $457$ (the data goes back to 2016-17). Let's now run it by season:

In [17]:
df.loc[A&B&C].groupby('season')[['hundred_spread_fade','hundred_ml_fade']].agg(['count','sum','mean'])

hundred_spread_fade                hundred_ml_fade              
                      count     sum   mean           count     sum  mean
season                                                                  
2016-17                  69  649.33   9.41              69 -469.82 -6.81
2017-18                  58  567.11   9.78              58  358.99  6.19
2018-19                  56  191.73   3.42              56 -236.82 -4.23
2020-21                  64 -754.17 -11.78              64 -310.88 -4.86
2021-22                  54 1003.00  18.57              54  289.94  5.37
2022-23                  51 -117.92  -2.31              51  295.12  5.79
2023-24                  49 1600.57  32.66              49 1757.17 35.86
2024-25                  51  272.16   5.34              51  324.41  6.36
2025-26                   5  264.01  52.80               5   97.83 19.57

This season is off to a good start. Let's look at who would have won us some money in this season so far.

In [18]:
df.loc[A&B&C,['date','team','opp','is_home','pts_diff_prev','pts_diff','pts','pts_opp','hundred_ml_fade','hundred_spread_fade']].set_index('date').tail(10)

team  opp  is_home  pts_diff_prev  pts_diff    pts  pts_opp  \
date                                                                     
2025-04-03  WAS  ORL     True              5    -12.00  97.00   109.00   
2025-04-09  BOS  ORL    False              2    -20.00  76.00    96.00   
2025-04-09  LAC  HOU     True              5     17.00 134.00   117.00   
2025-04-11  IND  ORL     True              2    -14.00 115.00   129.00   
2025-10-25  ATL  OKC     True              4    -17.00 100.00   117.00   
2025-10-27  CLE  DET    False              5     21.00 116.00    95.00   
2025-10-27  MIN  DEN     True              4    -13.00 114.00   127.00   
2025-11-01  BOS  HOU     True              1    -27.00 101.00   128.00   
2025-11-04  MIL  TOR    False              2    -28.00 100.00   128.00   
2025-11-05  NOP  DAL    False              4       NaN    NaN      NaN   

            hundred_ml_fade  hundred_spread_fade  
date                                              
2025-04-03             7.14              -100.00  
2025-04-09            28.57                90.91  
2025-04-09          -100.00              -100.00  
2025-04-11           135.00                90.91  
2025-10-25            28.57                90.91  
2025-10-27          -100.00              -100.00  
2025-10-27            42.55                95.24  
2025-11-01            52.63                86.96  
2025-11-04            74.07                90.91  
2025-11-05              NaN                  NaN

Rockin' with the theory, it says to **Fade the Pelicans** tonight against the **Mavericks**. If history repeats itself, your bet should return you a decent profit. 